### DISCUSSION TITLE
RPi4 + Sonoff Zigbee dongle + external HDD = interference?

### DISCUSSION THREAD

**Discussion by jrhansen** [Koenkk/zigbee2mqtt#11159] [General]
I have great trouble creating a reliable Zigbee setup with a Raspberry Pi 4 and a Sonoff Zigbee dongle. The symptoms are quite complex, but I think I have narrowed it down to following two things (still: sorry for the long post):

1. Plugging the dongle directly into the RPi4 results in the pairing of devices to fail. Looking at the log files while pairing, sometimes nothing at all happens, and sometimes the device announces itself and then leaves the network again by itself. I have found that using a USB extension cable for the dongle seems to make the Zigbee more reliable. Is there a general consensus that the RPi4 is noisy in terms of RF, and that the Sonoff dongle has to be placed further away from the RPi4? I have other setups with a cheap CC2531 dongle connected directly to a RPi4 with no problems, so maybe the Sonoff is more sensitive?

2. I am experimenting with booting the RPi4 from a simple USB flash drive as well as an HDD (not SSD) in an external enclosure with external power supply. The end goal is to use the external HDD, but for some reason the Zigbee network is still unstable, when using the HDD. When I boot from the USB flash drive, I get the more reliable Zigbee network described above, but with the HDD the Zigbee network is still bad even with the USB extension cable. There is another thread here indicating that other people also need to remove any external HDDs to make the Sonoff dongle work. Can anyone comment on this issue? Would it help to go to a SSD?

Note: I have 2.4 GHz WiFi at channel 4, and the zigbee network at channel 20, which should rule out interference between the two networks (?).

Thanks in advance :-)

**Comment by rikman122** [comment]
Hi @jrhansen 

I can confirm that I experienced issues using the Sonoff dongle directly connected to my RPi4. My setup also includes an external SSD connected to the 3.0 USB port and places right under the RPi. I tested with different USB ports and found the only solution was to use an extension cable. All the issues are now gone so, in my casa, RPi4 + SSD + Sonoff dongle with extension cable is working pretty stable.

Hope this help you.

**Reply by jrhansen** [reply]
At least it is one more reason (excuse) for me to get an SSD and replace the HDD. Still, I would like to understand if the HDD really causes the issues, given that replacing a 4 TB HDD with an SDD is quite expensive :-)

Thanks, @rikman122!

**Reply by Hedda** [reply]
No this EMI/EMF issue is with all USB 3.0 devices and all USB 3.x cables connected to a USB3 port so applies to both  SSD and HDD.

**Comment by lux73** [comment]
it seems that all Zigbee Coordinator had serious Problems with directly connected to USB

i have switched from RPi3 to APU Board - even there i have to use an USB Extension Cable...

**Reply by jrhansen** [reply]
As mentioned, I have used an ordinary CC2531 directly connected to an RPi4 without issues, so not all coordinators requires an extension cable - even though having one must be better. I choose to believe that the Sonoff dongle is especially prone to disturbances due to the receiver being very sensitive (or badly designed).

**Reply by lux73** [reply]
before i was using a cc2531 too - even with the old cc2531 i must use an USB Extension Cable with my RPi3 to get reliable Results

now i'm using cc2652rb - at first install i gave a try without extension but i had instantly droputs & massive delays without these Cable

so for me personally this cable is a must have :-)

**Comment by Hedda** [comment]
Electromagnetic fields around 2.4GHz is a known issue with USB 3.0 as tested and reported by Intel via this White Paper back in 2012:

https://www.usb.org/sites/default/files/327216.pdf

Recommendation is to connect Zigbee coordinator adapter to a USB 2.0 port or via USB 2.0 hub and also use a long USB extension cable.

If possible add electromagnetic shielding (ex. harddrive enclosure with grounded metal cover/casing) to USB 3.0 devices and USB 3.0 cables.

Also make sure use cables/wires with proper shielding (preferably what is sometimes called double-shielded USB cables.)

**Reply by jrhansen** [reply]
Thank you, @Hedda. This is VERY useful information. The Intel paper is quite scary though :-/

After providing some physical seperation between HDD and Sonoff cables, it seems things are working quite nicely. I still see one of the motion sensors announcing itself and then leaving again, but in the end the device paired correctly. I guess I will need to get my hands on some high quality shielded cables (short HDD/USB3 cable, longer USB extension cable), as well as built a setup, where the HDD enclosure can be properly grounded. I am an electrical engineer, so it should be possible :-)

Thanks again!

**Reply by Hedda** [reply]
> I am an electrical engineer, so it should be possible :-)

If you are an electrical engineer then you probably have access to some conductive metal tape and some Kapton tape as Intel did ;)

Alternatively, just buy one of those all-metal hard disk enclosures and an all-metal Raspberry Pi enclosure that looks like a heat sink.

**Comment by p4sI** [comment]
Having similar issues here. My Setup is a RPI4 with an external SSD connected on the USB3 port and an USB3 Hub connected to the USB2 port where my Sonoff Zigbee Stick is in. This works fine and the range is great. However, when I plug in an external HDD (Lenovo Thinkpad) into my MacBook which is about 50cm away from the RPI Setup the Zigbee network is disturbed in a way that its pretty much not working anymore at all...
So its also enough to have an external device / HDD close to your Zigbee stick and doesn't have to be in the same Raspberry, maybe this helps someone.

**Reply by Hedda** [reply]
Tip ia to buy a poweed USB 2.0 hub for the Zigbee dongle
